internal/handlers/gateway/modes.go

```diff
@@ -0,0 +1,31 @@
+package gateway
+
+import (
+	"fmt"
+
+	"github.com/os-observability/tempo-operator/apis/tempo/v1alpha1"
+)
+
+// ValidateModes validates the tenants mode specification.
+func ValidateModes(tempo v1alpha1.TempoStack) error {
```

Comment:
Isn't `ValidateTenants` more appropriate? 

Also shouldn't this be moved to the validation webhook? (or called from there as well)

Reply:
I guess calling it there too makes sense :)

Reply:
hm.. `ValidateTenantConfigs` then or `*Settings`?